cmd/mattermost/commands/config_test.go

```diff
@@ -158,6 +157,48 @@ func TestConfigSet(t *testing.T) {
 	})
 }
 
+func TestConfigReset(t *testing.T) {
+	th := Setup()
+	defer th.TearDown()
+
+	t.Run("No Error when no arguments are given (reset all the configurations)", func(t *testing.T) {
+		assert.NoError(t, th.RunCommand(t, "config", "reset"))
+	})
+
+	t.Run("No Error when a configuration section is given", func(t *testing.T) {
+		assert.NoError(t, th.RunCommand(t, "config", "reset", "JobSettings"))
+	})
+
+	t.Run("No Error when a configuration setting is given", func(t *testing.T) {
+		assert.NoError(t, th.RunCommand(t, "config", "reset", "JobSettings.RunJobs"))
+	})
+
+	t.Run("Error when the wrong configuration section is given", func(t *testing.T) {
+		assert.Error(t, th.RunCommand(t, "config", "reset", "InvalidSettings"))
+	})
+
+	t.Run("Error when the wrong configuration setting is given", func(t *testing.T) {
+		assert.Error(t, th.RunCommand(t, "config", "reset", "JobSettings.InvalidConfiguration"))
+	})
+
+	t.Run("Success when a configuration section is given", func(t *testing.T) {
```

Comment:
Can we add another test to verify the "reset all" affecting to multiple sections at the same time?

Reply:
Hey @jespino , since the "reset all" condition needs the user query to input "YES"  to confirm that they want to reset all of the configurations after inputting no further (zero) arguments, is there any way to simulate this user input as if there is a "YES" string input from the user when I ran the command at the test section? 

Thank you very much! :slightly_smiling_face:

Reply:
Hi @jespino , I am thinking of having an optional boolean flag perhaps called "--all" that when it's supplied by the value "true" will automatically reset all of the config (so that it can be run on the test) but if there's no argument at all (only "config reset"), then it will ask for user confirmation like [here](https://github.com/mattermost/mattermost-server/blob/531897b1f0d8176c1f983f921f1d1de618db0131/cmd/mattermost/commands/permissions.go#L66) and [here](https://github.com/mattermost/mattermost-server/blob/3716918c57b7b075676334ab4617ffea619704cc/cmd/mattermost/commands/team.go#L232). Looking forward to what you think! :slightly_smiling_face:

Reply:
Yes, I like the idea, but to be consistent with the rest of our CLI, I would suggest to use the '--confirm'  flag like in `cmd/mattermost/commands/reset.go`